src/main/java/com/ksm/wstbackoffice/controller/CountryController.java

```diff
@@ -29,7 +30,14 @@ public CountryController(CountryService countryService) {
 
     @GetMapping("{id}")
     public ResponseEntity<Country> findById(@PathVariable String id) {
+        if (!Pattern.matches("[0-9]{3}", id))
```

Comment:
As far as regexp is difficult to understand it's always good idea to create static field of the class and give it a meaningful name. In you case it could be privat static final String ONLY_NUMBERS_LENGHT3_REGEXP

Reply:
done